react/features/base/tracks/middleware.web.ts

```diff
@@ -207,3 +223,62 @@ function _removeNoDataFromSourceNotification({ getState, dispatch }: IStore, tra
         dispatch(trackNoDataFromSourceNotificationInfoChanged(jitsiTrack, undefined));
     }
 }
+
+/**
+ * Mutes or unmutes a local track with a specific media type.
+ *
+ * @param {Store} store - The redux store in which the specified action is
+ * dispatched.
+ * @param {Action} action - The redux action dispatched in the specified store.
+ * @param {MEDIA_TYPE} mediaType - The {@link MEDIA_TYPE} of the local track
+ * which is being muted or unmuted.
+ * @private
+ * @returns {void}
+ */
+function _setMuted(store: IStore, { ensureTrack, muted }: {
+    ensureTrack: boolean; muted: boolean; }, mediaType: MediaType) {
+    const { dispatch, getState } = store;
+    const localTrack = _getLocalTrack(store, mediaType, /* includePending */ true);
+    const state = getState();
+
+    if (localTrack) {
+        // The `jitsiTrack` property will have a value only for a localTrack for which `getUserMedia` has already
+        // completed. If there's no `jitsiTrack`, then the `muted` state will be applied once the `jitsiTrack` is
+        // created.
+        const { jitsiTrack } = localTrack;
+
+        if (jitsiTrack) {
+            setTrackMuted(jitsiTrack, muted, state, dispatch)
+            .catch(() => {
+                dispatch(trackMuteUnmuteFailed(localTrack, muted));
+            });
+        }
+    } else if (!muted && ensureTrack) {
+        // TODO(saghul): reconcile these 2 types.
+        const createMediaType = mediaType === MEDIA_TYPE.SCREENSHARE ? 'desktop' : mediaType;
+
+        typeof APP !== 'undefined' && dispatch(gumPending([ mediaType ], IGUMPendingState.PENDING_UNMUTE));
```

Comment:
typeof APP !== 'undefined' will always return true for web, can remove the check